src/types/multiModel.js

```diff
@@ -83,67 +73,62 @@ class MultiModel extends BaseType {
         throw new Error('MultiModel is virtual. It must be extended, and requestOne() must be overriden');
     }
 
-    requestMany (options: ?Object): void {
+    requestMany (options: Object): void {
         throw new Error('MultiModel is virtual. It must be extended, and requestMany() must be overriden');
     }
 
-    apply (response: any, options: ?Object): any {
-        let models = Array.isArray(response) ? response : [response];
-
-        const idKey = this.Model.idField;
-
-        const ids = models.map((modelData) => {
-            return modelData[idKey];
-        });
-
-        this.requestCache.set(options, ids);
-        this.store(models);
-    }
-
-    get (id: string, options: ?Object): any {
-        let model = this.fetch(id);
+    get (id: string, options: Object = {}): any {
+        let model = this._getModelById(id);
 
         if  (model) {
             return model;
 
         } else {
-            this.requestOne(id, options || {});
+            this.requestOne(id, options);
         }
     }
 
-    all (options: ?Object): Array<any> {
-        let ids = this.requestCache.get(options || {});
+    add (model: Object, index: ?number): Object {
 
-        if (ids) {
-            return ids.map((id) => {
-                return this.fetch(id);
-            });
-        } else {
-            this.requestMany(options || {});
-            return [];
+        if (!this.map) {
+            throw new Error('An unmapped OwnsMany cannot be added to');
         }
-    }
 
-    add (model: Object, index: number) {
-        if (this.map) {
-            this._parent.get(this.map).add(model.id, index);
-        }
+        this.map.add(model.id, index);
+
+        return this;
     }
 
     remove (model: Object) {
-        if (this.map) {
-            let mapIds = this._parent.get(this.map).all();
-            let removeIndex = mapIds.indexOf(model.id);
 
-            if (removeIndex >= 0) {
-                this._parent.get(this.map).remove(removeIndex);
-            }
+        if (!this.map) {
+            throw new Error(`An unmapped ${this.constructor.name} cannot be removed`);
+        }
+
+        const mapIds = this.map.all();
+        const removeIndex = mapIds.indexOf(model.id);
+
+        if (removeIndex >= 0) {
+            this.map.remove(removeIndex);
         }
     }
 
-    refresh () {
-        this.requestCache.clear();
+    move (model: Object, newIndex: number): Object {
```

Comment:
I think we should leave this in Multi-Model just in case someone has a use-case for it. It would be totally valid for an api to support this even if ours doesn't